### PR TITLE
Add support for building docker images with specified python version

### PR DESCRIPTION
**Pull request by agunapal**
## Description

Build docker images based on the specified python version
- Introduced an argument called `-py` for docker's `build.sh`
- Supported python versions are 3.8, 3.9 and 3.10
- Changed default Python version to 3.9

Fixes #([issue](https://github.com/pytorch/serve/issues/2145))

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Dev Docker Image with 3.10
```
 => [compile-image 3/3] RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.10 1     && update-alternatives --install /usr/local/bin/pip pip /usr/l  0.4s
 => [dev-image 1/2] RUN if [ "cpu" = "gpu" ]; then export USE_CUDA=1; fi     && git clone https://github.com/pytorch/serve.git     && cd serve     && git checkout maste  50.1s 
 => => #   Downloading numpy-1.24.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)                                                                        
 => => #      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.3/17.3 MB 103.4 MB/s eta 0:00:00                                                                                       
 => => # Requirement already satisfied: requests in /usr/lib/python3/dist-packages (from torchvision==0.14.1+cpu->-r requirements/torch_linux.txt (line 7)) (2.22.0)            
 => => # Collecting tqdm                                                                                                                                                        
 => => #   Downloading tqdm-4.64.1-py2.py3-none-any.whl (78 kB)                                                                                                                
 => => #      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.5/78.5 kB 19.0 MB/s eta 0:00:00    
```

- [ ] Default Docker image 
```
 => CACHED [runtime-image 2/9] RUN --mount=type=cache,target=/var/cache/apt     apt-get update &&     DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recomme  0.0s
 => CACHED [runtime-image 3/9] RUN useradd -m model-server     && mkdir -p /home/model-server/tmp                                                                          0.0s
 => CACHED [compile-image 2/8] RUN --mount=type=cache,id=apt-dev,target=/var/cache/apt     apt-get update &&     apt-get upgrade -y &&     apt-get install software-prope  0.0s
 => CACHED [compile-image 3/8] RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.9 1     && update-alternatives --install /usr/local/bin/pip pip   0.0s
 => CACHED [compile-image 4/8] RUN python$PYHON_VERSION -m venv /home/venv                                                                                                 0.0s
 => CACHED [compile-image 5/8] RUN python -m pip install -U pip setuptools                                                                                                 0.0s
 => CACHED [compile-image 6/8] RUN export USE_CUDA=1                                                                                                                       0.0s
 => [compile-image 7/8] RUN TORCH_VER=$(curl --silent --location https://pypi.org/pypi/torch/json | python -c "import sys, json, pkg_resources; releases = json.load(sys.  6.3s
 => => # Collecting pillow!=8.3.*,>=5.3.0                                                                                                                                      
 => => #   Downloading Pillow-9.4.0-cp39-cp39-manylinux_2_28_x86_64.whl (3.4 MB)                                                                                               
 => => #      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.4/3.4 MB 84.2 MB/s eta 0:00:00                                                                                        
 => => # Collecting requests                                                                                                                                                   
 => => #   Downloading requests-2.28.2-py3-none-any.whl (62 kB)                                                            
```

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: docker/build_image.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@ CUDA_VERSION=""
 UBUNTU_VERSION="ubuntu:20.04"
 USE_LOCAL_SERVE_FOLDER=false
 BUILD_WITH_IPEX=false
+PYTHON_VERSION=3.8
```
Comment (msaroufim) [inline docker/build_image.sh:14]: Seems like default is 3.8 and not 3.9 like docs indicate? Don't mind either way let's just stay consistent

File: docker/build_image.sh [hunk 2]
```diff
@@ -64,6 +66,17 @@ do
           BUILD_WITH_IPEX=true
           shift
           ;;
+        -py|--pythonversion)
+          PYTHON_VERSION="$2"
+          if [[ $PYTHON_VERSION = 3.8 || $PYTHON_VERSION = 3.9 || $PYTHON_VERSION = 3.10 ]]; then
+            echo "Valid python version"
+          else
+            echo "Valid python versions are 3.8, 3.9 and 3.10"
```
Comment (msaroufim) [inline docker/build_image.sh:74]: I think we can go ahead and support 3.11 as well, If someone is trying to torch.compile something with 3.11 they will get an error thrown from torch directly

**Review by msaroufim — APPROVED** [review]
Assuming all CI is green LGTM

@goha11 as FYI

File: docker/build_image.sh (hunk 1 above)
Comment (agunapal) [reply docker/build_image.sh]: Resolved. Thanks. I had not push the latest update

File: docker/build_image.sh (hunk 2 above)
Comment (agunapal) [reply docker/build_image.sh:74]: I will add docker support for Python 3.11 as a broader PR for TorchServe python 3.11 support

**Review by lxning — APPROVED** [review]

**Pull request merged**